src/main/java/fishron/Parser.java

```diff
@@ -0,0 +1,180 @@
+package fishron;
+
+/**
+ * The Parser class is responsible for parsing user input and converting it into executable commands.
+ */
+public class Parser {
+
+    /**
+     * Parses the user input and returns the corresponding Command object.
+     *
+     * @param input    The user input string.
+     * @param taskList The TaskList to operate on.
+     * @return A Command object representing the parsed command.
+     * @throws FishronException If there is an issue with parsing or an invalid command.
+     */
+    public static Command parse(String input, TaskList taskList) throws FishronException {
+
+        if (input.equalsIgnoreCase("bye")) {
+            return new ExitCommand();
+        }
+
+        isValidCommands(input, taskList);
+
+        if (input.equalsIgnoreCase("list")) {
+            return new ListCommand();
+
+        } else if (input.toLowerCase().startsWith("find")) {
+            String[] parts = input.split("find");
+            String keyword = parts[1].trim();
+            return new FindCommand(keyword);
+
+        } else if (input.toLowerCase().startsWith("mark")) {
+            int taskIndex = Integer.parseInt(input.split(" ")[1]);
+            return new MarkDoneCommand(taskIndex);
+
+        } else if (input.toLowerCase().startsWith("unmark")) {
+            int taskIndex = Integer.parseInt(input.split(" ")[1]);
+            return new UnmarkCommand(taskIndex);
+
+        } else if (input.toLowerCase().startsWith("delete")) {
+            int taskIndex = Integer.parseInt(input.split(" ")[1]);
+            return new DeleteCommand(taskIndex);
+
+        } else if (input.toLowerCase().startsWith("todo")) {
+            String[] parts = input.split("todo");
+            return new AddCommand(parseTodo(parts[1]));
+
+        } else if (input.toLowerCase().startsWith("deadline")) {
+            String[] parts = input.split("/by");
+            return new AddCommand(parseDeadline(parts[0].split("deadline")[1].trim(), parts[1].trim()));
+
+        } else if (input.toLowerCase().startsWith("event")) {
+            String[] parts = input.split("/from|/to");
+            return new AddCommand(parseEvent(parts[0].split("event")[1].trim(), parts[1].trim(), parts[2].trim()));
```

Comment:
Good use of trim() here. Avoids any user-generated error caused by additional spaces.